Symfony/CS/Fixer/Contrib/SingleQuoteFixer.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Gregor Harlan <gharlan@web.de>
+ */
+class SingleQuoteFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        foreach ($tokens as $token) {
+            if (!$token->isGivenKind(T_CONSTANT_ENCAPSED_STRING)) {
+                continue;
+            }
+
+            if (preg_match('/^"([^\'\\\\]*)"$/', $token->getContent(), $match)) {
```

Comment:
Not sure, the general advise is to not use a regex for detecting if a character is present as they are slower then string functions.

To be sure you would need to benchmark this :)